Kitodo/src/main/java/org/kitodo/services/data/base/SearchService.java

```diff
@@ -57,29 +58,34 @@
     private static final Logger logger = LogManager.getLogger(SearchService.class);
     protected Searcher searcher;
     protected Indexer indexer;
+    protected BaseType type;
 
     /**
      * Constructor necessary to use searcher in child classes.
      *
      * @param searcher
      *            for executing queries
      */
-    public SearchService(V dao, Searcher searcher) {
+    public SearchService(V dao, BaseType type, Indexer indexer, Searcher searcher) {
         super(dao);
         this.searcher = searcher;
+        this.indexer = indexer;
+        this.type = type;
     }
 
     /**
-     * Method saves document to the index of Elastic Search.
+     * Method converts JSON object object to DTO. Necessary for displaying in the
+     * frontend.
      *
-     * @param baseIndexedBean
-     *            object
+     * @param jsonObject
+     *            return from find methods
+     * @return DTO object
      */
-    public abstract void saveToIndex(T baseIndexedBean) throws CustomResponseException, IOException;
+    public abstract S convertJSONObjectToDTO(JSONObject jsonObject, boolean related) throws DataException;
```

Comment:
Missing JavaDoc for parameter `related`.

Reply:
added